metricbeat/module/elasticsearch/index/data_xpack.go

```diff
@@ -95,20 +100,47 @@ var (
 )
 
 func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, info elasticsearch.Info, content []byte) error {
-	var indicesStruct IndicesStruct
-	err := json.Unmarshal(content, &indicesStruct)
-	if err != nil {
-		return errors.Wrap(err, "failure parsing Indices Stats Elasticsearch API response")
-	}
-
 	clusterStateMetrics := []string{"metadata", "routing_table"}
 	clusterState, err := elasticsearch.GetClusterState(m.HTTP, m.HTTP.GetURI(), clusterStateMetrics)
 	if err != nil {
 		return errors.Wrap(err, "failure retrieving cluster state from Elasticsearch")
 	}
 
+	dec := json.NewDecoder(bytes.NewReader(content))
+
+	// read opening `{`
+	if _, err = dec.Token(); err != nil {
+		return errors.Wrap(err, errParse.Error())
+	}
```

Comment:
if doing so, we should still validate that the token we read matches our expectations. E.g. the jsonReader in the elasticsearch output has helpers like `ExpectDict` (checks token is `{`) or `ExpectArray` (checks token is `[`).